### ISSUE TITLE
Drop support for Ruby 2.0

### ISSUE THREAD

**Issue by zverok** [SciRuby/daru#336]
I believe we can allow ourselves to it now :)

This not only means specifying 2.1 as a minimum version in the Gemfile, but also switching from option hashes to proper keyword arguments everywhere, which will greatly enchance documentability and discoverability of features.

On this change, version should be bumped at least to 0.2, I believe.

PS: As a side note, I believe 0.x becames to be a bit odd for library of this size. Maybe we need something like "Roadmap to 1.0" somewhere?..

**Comment by v0dro** [comment]
Agreed. And yes, we should have a roadmap. Maybe we take inspiration from JRuby for this purpose? I've heard quite a few people talk fondly about their release schedules and roadmap.

**Comment by zverok** [comment]
Let's probably schedule the roadmap creation for August/September? When GSoC will be finished and we'll know where we are.

**Comment by v0dro** [comment]
Yes. Works.
